.github/workflows/bazel.yml

```diff
@@ -52,12 +52,41 @@ jobs:
           uses: actions/cache@v2
           with:
             path: ${{ github.workspace }}/.${{ env.BAZEL_CACHE }}
-            key: ${{ runner.os }}-${{ env.BAZEL_CACHE }}
+            key: ${{ runner.os }}-${{ env.BAZEL_CACHE }}-${{ github.sha }}
+            restore-keys: |
+              ${{ runner.os }}-${{ env.BAZEL_CACHE }}-
+
         - name: Bazel Cache Repo
           uses: actions/cache@v2
           with:
             path: ${{ github.workspace }}/.${{ env.BAZEL_CACHE_REPO }}
-            key: ${{ runner.os }}-${{ env.BAZEL_CACHE_REPO }}
+            key: ${{ runner.os }}-${{ env.BAZEL_CACHE_REPO }}-${{ github.sha }}
+            restore-keys: |
+              ${{ runner.os }}-${{ env.BAZEL_CACHE_REPO }}-
+
+        # This check is needed to ensure that Bazel's unbounded cache growth doesn't result in a
+        # situation where the cache never updates (e.g. due to exceeding GitHub's cache size limit)
+        # thereby only ever using the last successful cache version. This solution will result in a
+        # few slower CI actions around the time cache is detected to be too large, but it should
+        # incrementally improve thereafter.
+        - name: Ensure cache size
+          env:
+            BAZEL_CACHE_DIR: .${{ env.BAZEL_CACHE }}
+            BAZEL_CACHE_REPO_DIR: .${{ env.BAZEL_CACHE_REPO }}
+          run: |
+            # See https://stackoverflow.com/a/27485157 for reference.
+            EXPANDED_BAZEL_CACHE_PATH="${BAZEL_CACHE_DIR/#\~/$HOME}"
+            EXPANDED_BAZEL_CACHE_REPO_PATH="${BAZEL_CACHE_REPO_DIR/#\~/$HOME}"
+
+            CACHE_SIZE_MB=$(du -smc $EXPANDED_BAZEL_CACHE_PATH $EXPANDED_BAZEL_CACHE_REPO_PATH | grep total | cut -f1)
+            echo "Total size of Bazel cache (rounded up to MBs): $CACHE_SIZE_MB"
+            # Use a 4.5GB threshold since actions/cache compresses the results, and Bazel caches seem
+            # to only increase by a few hundred megabytes across changes for unrelated branches.
+            if [[ "$CACHE_SIZE_MB" -gt 4500 ]]; then
+              echo "Cache exceeds cut-off; resetting it (will result in a slow build)"
+              rm -rf $EXPANDED_BAZEL_CACHE_PATH
```

Comment:
don't you want to remove the EXPANDED_BAZEL_CACHE_REPO_PATH too here ?

Reply:
ye good catch!

Reply:
fixed @quentinDERORY